From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: <your git identity>
Date: Mon, 26 Oct 2026 13:10:22 +0000
Subject: [PATCH] patch: fix heap-buffer-overflow in unquote_file()

AddressSanitizer reports a heap-buffer-overflow in unquote_file()
(toys/posix/patch.c:287).

The change, in toys/posix/patch.c:

    -    if (!s[1]) error_exit("bad %s", filename);
    +    if (!*s || !s[1]) error_exit("bad %s", filename);

Fixes #572.
---
 toys/posix/patch.c | 2 +-
 1 file(s) changed, 1 insertion(s)(+), 1 deletion(s)(-)

diff --git a/toys/posix/patch.c b/toys/posix/patch.c
--- a/toys/posix/patch.c
+++ b/toys/posix/patch.c
@@ -281,7 +281,7 @@
 
   // quoted and escaped filenames are larger than the original
   for (t = newfile = xmalloc(strlen(s) + 1); *s != '"'; s++) {
-    if (!s[1]) error_exit("bad %s", filename);
+    if (!*s || !s[1]) error_exit("bad %s", filename);
 
     // don't accept escape sequences unless the filename is quoted
     if (*s != '\\') *t++ = *s;
--
leakproof package
